### PR TITLE
Handling Different User Types

### PR DESCRIPTION
**Pull request by celinechoiii**
## What's new in this PR
- Added genericUser type in types.tsx, along with RegularUser, Admin, and Employer interfaces
- Rewrote the functions in user.ts to handle three separate user collections
- Created new bottom tab navigator for employers

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Three new collections were added to firebase in replacement to the single User collection: regularUser, employer, and admin
- User functions were rewritten to parse, add, delete, and update users based on the access type
- The employer bottom tab stack includes Feed, Draft, and Signout

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"
- Please look over the code for the user.ts file to ensure its functionality

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- Review the parseUser function in user.ts to see if the user needs to be returned according to its access type
- Ensure the functionality of the getUser function for regularUser and employers
- Possible changes to the checkAndAddUser function

### Online sources
[//]: # "Optional - copy links to any tutorials or documentation that was useful to you when working on this PR"

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @surajr10 @themiachou8

### CODE REVIEW

File: src/navigation/employerStack.tsx
Comment (surajr10) [inline src/navigation/employerStack.tsx:1]: i'd recommend rebasing from main because part of the navigation folder structure changed as we now have a folder for bottomTabNavigators which this file would fall under.

File: src/types/types.tsx [hunk 1]
```diff
@@ -1,14 +1,24 @@
-export interface User {
+export interface genericUser {
   id: string;
   access: string;
-  createdJobs: string[]; // switched to string of jobIds to match Firebase
   email: string | null;
```
Comment (surajr10) [inline src/types/types.tsx]: I think only admins will have an email field

File: src/types/types.tsx [hunk 2]
```diff
@@ -1,14 +1,24 @@
-export interface User {
+export interface genericUser {
   id: string;
   access: string;
```
Comment (surajr10) [inline src/types/types.tsx:3]: I think this field will be obsolete with our new schema as it used to represent the user's type

File: src/firebase/firestore/user.ts [hunk 3]
```diff
@@ -9,58 +9,81 @@ import {
...
-  const docSnap = await getDoc(docRef);
-  if (docSnap.exists()) {
-    console.log('User data:', docSnap.data());
-    return parseUser(docSnap);
-  }
-  // doc.data() will be undefined in this case
-  console.log('No such document!');
-  return null;
+export const getUser = async (id: string): Promise<genericUser | null> => {
+  const userRefs = [
```
Comment (surajr10) [inline src/firebase/firestore/user.ts]: i would say you can extract this from the function and keep it as a constant at the top of the file. what might be a good practice is defining constants for each of the collection names and then also defining the constant userRefs which is the list of refs as you have here

File: src/firebase/firestore/user.ts [hunk 4]
```diff
@@ -9,58 +9,103 @@ import {
...
+  [<Employer>{}, EMPLOYER_COLLECTION_NAME],
+])
+
+// QueryDocumentSnapshot<DocumentData>
 const parseUser = async (document: QueryDocumentSnapshot<DocumentData>) => {
   const userId = document.id.toString();
   const data = document.data();
+  const type = data.access; 
+
+  // if (!userTypeToConstructorMap.has(type)) {
```
Comment (surajr10) [inline src/firebase/firestore/user.ts]: remove commented chunk of code

**Review by surajr10 — APPROVED** [review]
